### PR TITLE
support python 3.11

### PR DESCRIPTION
**Pull request by cosmicBboy**
# TL;DR

Fixes https://github.com/flyteorg/flyte/issues/3312. This PR makes the necessary changes to support python 3.11

It also addresses https://github.com/flyteorg/flyte/issues/3166, removing the old numpy dtype aliases.

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description

See TLDR

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3312

## Follow-up issue
_NA_

### CODE REVIEW

File: plugins/flytekit-dbt/requirements.in [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .
 -e file:.#egg=flytekitplugins-dbt
 # dbt-sqlite and dbt-core should be compatible
-dbt-core==1.2.1
+dbt-core==1.4.1
 dbt-sqlite==1.2.0a2
```
Comment (eapolinario) [inline plugins/flytekit-dbt/requirements.in:5]: we should remove `dbt-core` and just list `dbt-sqlite` in this file (this should bring a version of `dbt-core` that's compatible).

File: plugins/flytekit-greatexpectations/setup.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 microlib_name = f"flytekitplugins-{PLUGIN_NAME}"
 
-plugin_requires = ["flytekit>=1.3.0b2,<2.0.0", "great-expectations>=0.13.30", "sqlalchemy>=1.4.23"]
+plugin_requires = ["flytekit>=1.3.0b2,<2.0.0", "great-expectations>=0.13.30", "sqlalchemy>=1.4.23,<2.0.0"]
```
Comment (wild-endeavor) [inline plugins/flytekit-greatexpectations/setup.py:7]: needed?  or just nice to have?

**Review by wild-endeavor — APPROVED** [review]

File: plugins/flytekit-greatexpectations/setup.py (hunk 2 above)
Comment (cosmicBboy) [reply plugins/flytekit-greatexpectations/setup.py:7]: needed, since GE doesn't support sqlalchemy 2+ yet

**Review by wild-endeavor — APPROVED** [review]

**Comment by cosmicBboy** [comment]
Note: this may also address https://github.com/flyteorg/flyte/issues/2115 (?)

**Review by wild-endeavor — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]
Thank you!

**Review by pingsutw — APPROVED** [review]

**Pull request merged**